test/lit/isorecursive-good.wast

```diff
@@ -7,66 +7,82 @@
   (rec
     ;; CHECK:      (rec
     ;; CHECK-NEXT:  (type $super-struct (struct (field i32)))
-    (type $super-struct (struct i32))
+    (type $super-struct (sub (struct i32)))
```

Comment:
Why did this change?

Reply:
This is to test the new parsing logic that allows using `sub` without specifying any supertype.